test/models/content_view_version_test.rb

```diff
@@ -138,6 +138,42 @@ def test_components_needing_errata
       assert_includes @composite_version.components_needing_errata([errata]), component
     end
 
+    def test_check_ready_to_promote!
+      dev = KTEnvironment.find(katello_environments(:dev))
+      composite_version = ContentViewVersion.find(katello_content_view_versions(:composite_view_version_1))
+
+      # version with no envs
+      library_view_version_1 = ContentViewVersion.find(katello_content_view_versions(:library_view_version_1))
+      # version in library & dev
+      library_dev_view_version = ContentViewVersion.find(katello_content_view_versions(:library_dev_view_version))
+
+      # tests where the component view versions must exist in target environment
+      restrict_composite_view = Setting.create(:name => 'restrict_composite_view', :category => 'Setting::Katello',
+                                               :settings_type => 'boolean', :default => true)
+
+      composite_version.content_view.components = [library_view_version_1]
+      composite_version.save!
+      assert_raises RuntimeError do
+        composite_version.check_ready_to_promote!(dev)
+      end
+
+      composite_version.content_view.components = [library_dev_view_version]
+      composite_version.save!
+      assert composite_version.check_ready_to_promote!(dev)
+
+      # tests where the component view versions do not need to exist in target environment
+      restrict_composite_view.default = false
+      restrict_composite_view.save!
+
+      composite_version.content_view.components = [library_view_version_1]
+      composite_version.save!
+      assert composite_version.check_ready_to_promote!(dev)
+
+      composite_version.content_view.components = [library_dev_view_version]
+      composite_version.save!
+      assert composite_version.check_ready_to_promote!(dev)
```

Comment:
This test appears to be covering multiple conditions which we would normally split up into individual test functions isolated to those scenarios.